Python/pystate.c

```diff
@@ -308,6 +311,13 @@ PyInterpreterState_Clear(PyInterpreterState *interp)
     if (_PyRuntimeState_GetFinalizing(runtime) == NULL) {
         _PyWarnings_Fini(interp);
     }
+
+    /* We don't clear sysdict and builtins until the end of this function.
+       Because clearing other attributes can execute arbitrary Python code
+       which reuqires sysdict and builtins. */
+    PyDict_Clear(sysdict);
```

Comment:
I don't see the point of adding two local variables, why not accessing directly the structure member?

```suggestion
    PyDict_Clear(interp->sysdict);
```

Same remark on folowing line.

Reply:
interp->sysdict and interp->builtins will be cleared before this line. https://github.com/python/cpython/blob/master/Python/pystate.c#L297-L298

Reply:
Aha, I didn't notice. In this case, I suggest to move Py_CLEAR(interp->sysdict) and  Py_CLEAR(interp->builtins) at the end. Something like:

```
PyDict_Clear(interp->sysdict);
PyDict_Clear(interp->builtins);
Py_CLEAR(interp->sysdict)
Py_CLEAR(interp->builtins);
```

Reply:
updated. And I delete the description info.